src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectInfo.cs

```diff
@@ -618,7 +618,8 @@ internal Type[] GetMemberTypes(string[] inMemberNames, Type objectType)
             bool isFound = false;
             for (int i = 0; i < _cache._memberInfos.Length; i++)
             {
-                if (!memberMissing && inMemberNames[i].Equals(_cache._memberInfos[i].Name))
+                if (!memberMissing && 
+                    inMemberNames[i].Equals(FormatterServices.GetFullMemberInfoName(_cache._memberInfos[i], objectType)))
```

Comment:
I see. Your killer argument is FormatterServices.GetSerializableMembers which is public. So we NEED to change that :/